### ISSUE TITLE
ENH: seasonal_decompose for 2-D 

### ISSUE THREAD

**Issue by josef-pkt** [statsmodels/statsmodels#2820]
The seasonal_decompose function works only for 1-dimensional data. All the unit test cases are 1-D.

see https://github.com/statsmodels/statsmodels/issues/2733#issuecomment-185157826 for report/question and current exception raised.
duplicate issue with more description #2821

Based on briefly browsing the code, I think the function can be made to work for 2-D.

AFAICS:
- convolution_filter already works for 2-D on axis=0, according to docstring
- seasonal mean: calculates the mean over all columns for a seasonal period, i.e. common seasonality
- period_averages and creating seasonal assumes 1-d

**Comment by josef-pkt** [comment]
I got it to work on a numpy example just adding some `axis=0`, and 2d tiling.

(I grabbed a random python on the wrong computer to edit. PR coming up)

**Comment by Irene-GM** [comment]
Can I see the solution to the problem? Not sure on how (and where) to use the 2D tiling plus axis=0. Do you have a dummy test example you can share? Thanks a lot! :-)

**Comment by josef-pkt** [comment]
Soon I have a bit of a mess right now because I made the change on top of another old working branch.

My example to try it out is very simple and just uses numpy

```
import numpy as np
from statsmodels.tsa.seasonal import seasonal_decompose


nobs = 10 * 12
x = np.random.randn(nobs, 4)
s = np.tile((np.arange(12) - 6)**2, 10)
y = s[:,None] + x

res = seasonal_decompose(y[:,:], freq=12)
res0 = seasonal_decompose(y[:,0], freq=12)
print(res0.resid - res.resid[:,0])
```

**Comment by josef-pkt** [comment]
here are the changes. https://github.com/statsmodels/statsmodels/compare/master...josef-pkt:seasonal_2d

Note, this was very quickly done and only checked on the example.

**Comment by Irene-GM** [comment]
Thanks Josef! I tried and it works like a charm also with my example. It nicely leaves the result of the 34 decompositions in a dataframe organized per columns, so now I can continue with my experiments with the seasonal components of the time-series. I am very grateful for your help and support, because the alternative was using R (RStudio) and its a platform "isolated" of my Python workflows, so I preferred to keep pythonic, like this. Kudos! :-)

**Comment by josef-pkt** [comment]
merged as part of #3031
